### PR TITLE
Tag health metrics with service, env, and version

### PR DESCRIPTION
**Pull request by randomanderson**
Adds service, env, and version tags to health metrics

### CODE REVIEW

File: dd-trace-core/src/main/java/datadog/trace/core/CoreTracer.java [hunk 1]
```diff
@@ -568,16 +569,27 @@ private static StatsDClient createStatsDClient(final Config config) {
...
+          statsdTag(LANG_INTERPRETER_VENDOR_STATSD_TAG, DDTraceCoreInfo.JAVA_VM_VENDOR));
+      constantTags.add(statsdTag(TRACER_VERSION_STATSD_TAG, DDTraceCoreInfo.VERSION));
+      constantTags.add(statsdTag("service", config.getServiceName()));
+
+      final String version = config.getMergedSpanTags().get(GeneralConfig.VERSION);
+      if (version != null && !version.isEmpty()) {
+        constantTags.add(statsdTag("version", version));
+      }
+
+      final String env = config.getMergedSpanTags().get(GeneralConfig.ENV);
```
Comment (bantonsson) [inline dd-trace-core/src/main/java/datadog/trace/core/CoreTracer.java:586]: Not a big thing, but could you call `config.getMergedSpanTags()` only once? It actually creates a new `Map` and merge the tags every time it's called.

File: dd-trace-core/src/main/java/datadog/trace/core/CoreTracer.java [hunk 2]
```diff
@@ -568,16 +569,27 @@ private static StatsDClient createStatsDClient(final Config config) {
...
+      if (version != null && !version.isEmpty()) {
```
Comment (bantonsson) [inline dd-trace-core/src/main/java/datadog/trace/core/CoreTracer.java:587]: Is this `version` supposed to be `env`?

**Review by bantonsson — COMMENTED** [review]
👍

**Comment by richardstartin** [comment]
Does this need a test? I.e. is it something important that we rely on being correct?

**Review by tylerbenson — APPROVED** [review]

File: dd-trace-core/src/test/groovy/datadog/trace/core/CoreTracerTest.groovy [hunk 3]
```diff
@@ -64,6 +67,62 @@ class CoreTracerTest extends DDSpecification {
     tracer.statsDClient instanceof NonBlockingStatsDClient
   }
 
+  def "verify service, env, and version are added as stats tags"() {
```
Comment (richardstartin) [inline dd-trace-core/src/test/groovy/datadog/trace/core/CoreTracerTest.groovy:70]: nice

**Pull request merged**